src/internal/utils/bucketHelper.ts

```diff
@@ -43,19 +44,26 @@ import { UICONFIG } from "../constants/uiConfig";
 import { getTranslation } from "./translations";
 
 export function sanitizeFilters(newReferencePoint: IExtendedReferencePoint): IExtendedReferencePoint {
-    const usedAttributeBucketItems = getAllAttributeItems(newReferencePoint.buckets);
+    const attributeBucketItems = getAllAttributeItems(newReferencePoint.buckets);
+    const measureBucketItems = getAllMeasureItems(newReferencePoint.buckets);
 
     const filteredFilters = newReferencePoint.filters.items.filter((filterBucketItem: IFiltersBucketItem) => {
-        const hasCorrespondingBucketItem = usedAttributeBucketItems.some(
-            (attributeBucketItem: IBucketItem) => {
-                const filter = filterBucketItem.filters[0];
-                if (isAttributeFilter(filter) || isDateFilter(filter)) {
-                    return attributeBucketItem.attribute === filter.attribute;
-                }
-                return false;
-            },
-        );
-        return filterBucketItem.autoCreated === false || hasCorrespondingBucketItem;
+        const filter = filterBucketItem.filters[0];
+
+        if (isAttributeFilter(filter) || isDateFilter(filter)) {
+            const hasCorrespondingBucketItem = attributeBucketItems.some(
+                (attributeBucketItem: IBucketItem) => attributeBucketItem.attribute === filter.attribute,
+            );
+            return filterBucketItem.autoCreated === false || hasCorrespondingBucketItem;
+        } else if (isMeasureValueFilter) {
+            const hasCorrespondingBucketItem = measureBucketItems.some(
+                (measureBucketItem: IBucketItem) =>
+                    measureBucketItem.localIdentifier === filter.measureLocalIdentifier,
+            );
+            return attributeBucketItems.length !== 0 && hasCorrespondingBucketItem;
```

Comment:
Maybe `attributeBucketItems.length > 0` would be more readable?

Reply:
OK, done